docs/quickstart.md

```diff
@@ -0,0 +1,173 @@
+
+If you don't already have access to a lakeFS server, you can quickly start a local instance using Docker Compose with a [configuration file](https://github.com/appliedAI-Initiative/lakefs-spec/blob/main/hack/docker-compose.yml) provided in the `lakefs-spec` repository:
+
+```shell
+$ curl https://raw.githubusercontent.com/appliedAI-Initiative/lakefs-spec/main/hack/docker-compose.yml | docker-compose -f - up
+```
+
+If you do not have `curl` installed on your machine or would like to examine and/or customize the container configuration, you can also create a `docker-compose.yml` file locally and use it with `docker-compose up`:
+
+```yaml
+--8<-- "https://raw.githubusercontent.com/appliedAI-Initiative/lakefs-spec/main/hack/docker-compose.yml:3:"
+```
+
+In order to allow `lakefs-spec` to automatically discover credentials to access this lakeFS instance, create a `.lakectl.yaml` in your home directory containing the following:
+
+```yaml
+credentials:
+  access_key_id: AKIAIOSFOLQUICKSTART
+  secret_access_key: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
+server:
+  endpoint_url: http://127.0.0.1:8000
+```
+
+After the container has finished initializing, you can access the [web UI](http://localhost:8000) of your local lakeFS deployment in your browser. Fill out the setup form, where you can optionally share your email address with the developers of lakeFS to receive updates on their product. Next, you can log into your fresh lakeFS instance with the credentials listed above.
+
+!!! success
+    Your fresh local lakeFS instance is a playground for you to explore lakeFS functionality. 
+    
+    In the next step, we will create your first repository on this server.
+
+### Create a lakeFS repository
+
+Once you have logged into the web UI of the lakeFS server for the first time, you can create an empty repository on the next page.
+Click the small _Click here_ link at the bottom of the page to proceed and create a repository named `repo` (we don't want to add the sample data for this guide):
+
+![](_images/quickstart-lakefs-sample-repo.png)
+
+??? tip "Tip: Creating a repository later"
+
+    If you have inadvertently skipped over the quickstart repository creation page, you can always create a new repository on the [_Repositories_ tab](http://localhost:8000/repositories) in the lakeFS web UI (and optionally choose to add the sample data):
+
+    ![](_images/quickstart-lakefs-repositories.png)
+
+!!! success
+    You have successfully created a lakeFS repository named `repo`, ready to be used with `lakefs-spec`.
+
+### Using the lakeFS `fsspec` file system
+
+While the `quickstart` repository already contains some files, we won't be using them for the remainder of this guide. Instead, we will now use the `lakefs-spec` file system interface to upload a file to the repository we just created, make a commit, and read back the committed data.
+
+To get started, create a file called `quickstart.py` with the following contents:
```

Comment:
Is the intention that a user creates quickstart.py, adds pieces of code to this file after reading each step of this doc , and runs this script (in a command line) each time after adding piece of code/after each step? Then maybe a notebook more convenient (while following this doc, I ran the code piece by piece in the notebook).

Reply:
Good catch! That was a leftover from a previous revision of the guide.

I've rewritten the introduction to this section in 8c15872d99b8d067a29833bc655dc8a0b2643bb8, also giving some more guidance on how to follow along with the examples and what to do with the code snippets:

![image](https://github.com/appliedAI-Initiative/lakefs-spec/assets/322624/ebfdee08-b4dc-46e9-bd69-aaefa0c6a92c)